concurrency_limiter_test.go

```diff
@@ -0,0 +1,207 @@
+package main
```

Comment:
Could you run go test on this with -count and a > 1 number?

Reply:
Yeah, I've been testing with:
`go test -race -count=1000 ./concurrency-limiter/`

Reply:
That would be great to add in `How Has This Been Tested?` in any future PRs

Reply:
Good point.